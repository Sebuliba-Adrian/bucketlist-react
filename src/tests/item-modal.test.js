import React from 'react';
import renderer from 'react-test-renderer';
import ItemModal from '../components/items/item-modal';


const props = {
  theId: 1,
  title: 'Create item',
  action: 'Create',
  selectedBucketlist: {
    id: 1,
    title: 'Selected Bucketlist',
    description: 'This is the selected bucketlist',
    created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
  },
  item: {
    id: 1,
    title: 'Test Bucket',
    description: 'This is a test bucket',
    created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    status: false,
  },
  request: jest.fn(),
};
it('renders correctly', () => {
  const itemModal = renderer.create(<ItemModal {...props} />).toJSON();
  expect(itemModal).toMatchSnapshot();
});
