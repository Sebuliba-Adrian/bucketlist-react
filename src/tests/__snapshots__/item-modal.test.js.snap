// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="modal fade"
  id={1}
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLabel"
        >
          Create item
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <label
          htmlFor="tiltle"
        >
          Title
        </label>
        <div
          className="input-group"
          onChange={[Function]}
        >
          <input
            className="form-control"
            id="tiltle"
            name="title"
            onChange={[Function]}
            type="text"
            value="Test Bucket"
          />
        </div>
        <br />
        <label
          htmlFor="description"
        >
          Description
        </label>
        <div
          className="input-group"
        >
          <textarea
            className="form-control"
            id="description"
            name="description"
            onChange={[Function]}
            type="text"
            value="This is a test bucket"
          />
        </div>
        <br />
        <div
          className="form-group"
        >
          <label
            htmlFor="status"
          >
            Status
          </label>
          <select
            className="form-control"
            id="status"
            name="status"
            onChange={[Function]}
            value="0"
          >
            <option
              value="1"
            >
              Finished
            </option>
            <option
              value="0"
            >
              Pending
            </option>
          </select>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Create
        </button>
      </div>
    </div>
  </div>
</div>
`;
